frontend/src/features/admin-form/responses/ResponsesPage.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router'
+import { Skeleton } from '@chakra-ui/react'
+
+import { StorageModeSubmissionMetadata } from '~shared/types/submission'
+
+import Button from '~components/Button'
+
+import { useAdminFormSettings } from '../settings/queries'
+
+import { useFormResponses } from './queries'
+import useDecryptionWorkers from './useDecryptionWorkers'
+
+const ResponsesPage = (): JSX.Element => {
+  const { data: settings } = useAdminFormSettings()
+  const { data, isLoading } = useFormResponses()
+  const { formId } = useParams()
+  const [secretKey, setSecretKey] = useState<string>('')
+  const { downloadEncryptedResponses } = useDecryptionWorkers()
+
+  return (
+    <Skeleton isLoaded={!isLoading && !!data}>
+      <div>
+        Enter secret key:
+        <input
+          style={{ backgroundColor: 'grey' }}
+          type="text"
+          value={secretKey}
+          onChange={(e) => setSecretKey(e.target.value)}
+        />
+        <Button
+          onClick={() =>
+            downloadEncryptedResponses(formId!, settings!.title, secretKey)
```

Comment:
I think we should extract a new `handleExportCsv` function that first checks whether the formId and settings exist instead of using `!` to forcibly assert truthiness; this essentially defeats the point of typescript;

```tsx
const handleExportCsv = useCallback(() => {
  if (!formId || !settings?.title) return
  return downloadEncryptedResponses(formId, settings.title, secretKey)
}, [])
```